src/app/api/tools/cowswap/orderFlow.ts

```diff
@@ -91,10 +94,42 @@ export async function orderRequestFlow({
   console.log("Built Order", order);
 
   const orderUid = await orderbook.sendOrder(order);
-  const orderUrl = orderbook.getOrderLink(orderUid);
-  console.log("Order Posted", orderUrl);
+  console.log("Order Posted", orderbook.getOrderLink(orderUid));
+
+  const sellTokenDecimals = isNativeAsset(quoteRequest.sellToken) ? 18 : 18; // Default to 18, replace with actual fetch
+  const buyTokenDecimals = 18; // Default to 18, replace with actual fetch
```

Comment:
In the parse file, you can return `buyTokenData & sellTokenData`. They contain the decimals.

You can extend the return type here https://github.com/bh2smith/near-cow-agent/blob/9b13787e4e4c4c78d332cc01c5ae9de73729a09b/src/app/api/tools/cowswap/util/parse.ts#L25


`ParsedQuoteRequest & {buyTokenData: TokenData, sellTokenData: TokenData }`

then return this and use it here.